improver_tests/wxcode/wxcode/test_utilities.py

```diff
@@ -518,6 +519,15 @@ def test_check_tree(modify_tree, expected):
     assert result == expected
 
 
+def test_check_tree_unreachable_leaf():
+    """Add an unreachable leaf and check for an error."""
+    expected = "Unreachable node 'kittens'"
+    tree = wxcode_decision_tree()
+    tree["kittens"] = tree["lightning"].copy()
+    result = check_tree(tree)
+    assert result == expected
+
+
```

Comment:
This test could be replaced with an extension of the above parameterisation, e.g.
```
        (
            "mist_conditions",
            "if_true",
            "no_precipitation_cloud",
            "Unreachable node 'fog_conditions'",
        ),
```

Reply:
Good idea. I didn't think of this. Now done.